File: src/pe-003.rb
x = 13195

def find_factors num
	factors = []
	if num%2 == 0
		num/=2
	end
	for i in 2...num
		if num%i == 0
			factors.push(i)
		end
	end	
	return factors
end

def check_prime primeno
	if find_factors(primeno)==[]
		return true
	else
		return false
	end
end

#find_factors(13195).reverse_each do |factor|
find_factors(600851475143).reverse_each do |factor|
	if check_prime(factor)
		puts factor
	end
end

File: src/pe-003.js
function checkPrime(num) {
    for (x=2;x<num;x++) {
        if (num%x==0) {
            return false;
        }
    }
    return true;
}

function checkFactor(num, factor) {
    if (num%factor==0) {
        return true;
    }
    return false;
}

function largestPrime(num) {
    numroot = Math.floor(Math.sqrt(num));
    for (i=numroot;i>0;i--) {
        if (checkFactor(num, i) && checkPrime(i)) {
            return i;
        }
    }
    return false;
}

console.log(largestPrime(600851475143));